validator/client/validator.go

```diff
@@ -105,8 +105,8 @@ func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.UpdateAssignments")
 	defer span.Finish()
 
-	if slot%params.BeaconConfig().EpochLength != 0 {
-		// Do nothing if not epoch start.
+	if slot%params.BeaconConfig().EpochLength != 0 && v.assignment != nil {
```

Comment:
This should be `||`, why wait until an epoch start if you don't have your assignments yet?

Reply:
Resolved offline, this is the exit condition and I misunderstood